### PR TITLE
copr_v2: Enhance raw coprocessor protocol and API

### PR DESCRIPTION
**Pull request by andylokandy**
### What problem does this PR solve?

Issue Number: #9747

### What is changed and how it works?

- Update protobuf to https://github.com/pingcap/kvproto/pull/765
- Rename `coprocessor_v2` request to `raw_coprocessor`
- Unify `StorageError` and `PluginError`
- Feed plugin key ranges instead of `Region`
- Other refactor

### Release note 

- No release note

### CODE REVIEW

**Comment by andylokandy** [comment]
PTAL @Jibbow

**Comment by skyzh** [comment]
/retitle coprv2: Enhance raw coprocessor protcol and API

**Comment by skyzh** [comment]
/title coprv2: Enhance raw coprocessor protocol and API

**Review by skyzh — COMMENTED** [review]
LGTM generally. Will approve after kvproto PR is merged. By the way, why passing `Vec<Range<Key>>` instead of `&[Range<Key>]` to the plugin?

**Comment by andylokandy** [comment]
> By the way, why passing Vec<Range<Key>> instead of &[Range<Key>] to the plugin?

Because it will invoke raw coprocessor once per request, so there is no need to keep the ownership.

File: src/server/metrics.rs
Comment (Jibbow) [inline src/server/metrics.rs:52]: In some lines, we now have `raw_coprocessor`, in others `copr_v2`. Is this confusing?

**Review by Jibbow — APPROVED** [review]
Thank you! LGTM

**Review by zhongzc — APPROVED** [review]

File: src/server/metrics.rs
Comment (andylokandy) [reply src/server/metrics.rs:52]: `copr_v2` plugin is able to handle both `raw_coprocessor` and `txn_coprocessor`. So at the request level, I named it `RawCoprocessor`, and at the plugin level, it's called `copr_v2`.

**Review by skyzh — APPROVED** [review]

File: src/server/metrics.rs
Comment (Jibbow) [reply src/server/metrics.rs:52]: makes sense! thanks!

**Comment by skyzh** [comment]
Seems that the CI fails constantly. We should investigate more before merging.

**Comment by skyzh** [comment]
💣💣💣

**Comment by skyzh** [comment]
cc @zhongzc seems that you need to lgtm again :(

**Comment by andylokandy** [comment]
@Jibbow Would you like to `/lgtm` again?

**Pull request merged**
